tests/unit/sources/test_sources.py

```diff
@@ -45,24 +93,55 @@ def test_get_source_handler_class_with_invalid_type():
 @pytest.mark.parametrize(
     ("source", "result"),
     [
+        (".", "local"),
+        (".7z", "7z"),
+        (".deb", "deb"),
+        (
+            "http://archive.ubuntu.com/ubuntu/pool/main/g/glibc/libc6_2.39-0ubuntu8.3_amd64.deb",
+            "deb",
+        ),
+        (".rpm", "rpm"),
+        (".snap", "snap"),
         (".tar.gz", "tar"),
         (".tar.bz2", "tar"),
+        (".tar.zst", "tar"),
         (".tgz", "tar"),
         (".tar", "tar"),
         ("git:", "git"),
         ("git@", "git"),
         ("git+ssh:", "git"),
+        ("git+https:", "git"),
         (".git", "git"),
-        ("lp:", "bzr"),
-        ("bzr:", "bzr"),
-        ("svn:", "subversion"),
```

Comment:
These were leftovers in `get_source_type_from_uri` - craft-parts doesn't have these source handlers. While this is technically a backwards-incompatible change for `get_source_type_from_uri`, all practical use cases are that this would have errored when trying to find a source class anyway.